mission_control/test/test_depth_heading_behavior.py

```diff
@@ -0,0 +1,166 @@
+        self.depth_heading_goal = []
+
+        # Subscribers
+        self.exec_state_sub = rospy.Subscriber('/mission_control_node/report_mission_execute_state',
+                                               ReportExecuteMissionState, self.callback_mission_execute_state)
+
+        self.load_state_sub = rospy.Subscriber('/mission_control_node/report_mission_load_state',
+                                               ReportLoadMissionState, self.callback_mission_load_state)
+
+        self.report_missions_sub = rospy.Subscriber('/mission_control_node/report_missions',
+                                                    ReportMissions, self.callback_report_mission)
+
+        self.depth_heading_msg = rospy.Subscriber('/mngr/depth_heading',
+                                                 DepthHeading, self.callback_publish_depth_heading_goal)
+
+        # Publishers
+        self.simulated_mission_control_load_mission_pub = rospy.Publisher('/mission_control_node/load_mission',
+                                                                          LoadMission, latch=True, queue_size=1)
+
+        self.simulated_mission_control_execute_mission_pub = rospy.Publisher(
+            '/mission_control_node/execute_mission', ExecuteMission, latch=True, queue_size=1)
+        
+        self.simulated_pose_estimator_pub = rospy.Publisher('/pose/corrected_data',
+                                                            CorrectedData, queue_size=1)
+
+    def callback_mission_execute_state(self, msg):
+        self.report_execute_mission = msg
+
+    def callback_report_mission(self, msg):
+        self.report_mission = msg
+
+    def callback_mission_load_state(self, msg):
+        self.mission_load_state = msg.load_state
+
+    def callback_publish_depth_heading_goal(self, msg):
+        self.depth_heading_goal = [msg.depth,
+                                  msg.heading,
+                                  msg.speed_knots,
+                                  msg.ena_mask]
+
+    def test_mission_with_depth_heading_behavior(self):
+
+        self.mission_load_state_flag = False
+        mission_to_load = LoadMission()
+        mission_to_load.mission_file_full_path = self.dir_path + "test_missions/depth_heading_mission_test.xml"
+        self.simulated_mission_control_load_mission_pub.publish(mission_to_load)
+
+        def load_mission():
+            return self.mission_load_state == ReportLoadMissionState.SUCCESS
+        self.assertTrue(wait_for(load_mission),
+                        msg='Mission control must report SUCCESS')
+        rospy.loginfo("Mission Loaded")
+
+        mission_to_execute = ExecuteMission()
+        mission_to_execute.mission_id = 1
+        self.simulated_mission_control_execute_mission_pub.publish(
+            mission_to_execute)
+        rospy.loginfo("execute msg")
+
+        def depth_heading_goals_are_setted():
+            return self.depth_heading_goal == [1.0, 2.0, 3.0, 7]
```

Comment:
@patapufate nit: checking the full message would read better e.g. `self.goal.depth == x` rather than `self.goal == [x]`

Reply:
fixed